### PR TITLE
Fix 1594

### LINKED ISSUE #1594: Allow Empty records in custom functions
Follow up to #1490. 

Allow empty records to receive any record.

### PR DESCRIPTION
**Pull request by MikeStall**
Tactical fix #1594 . 

Need to let Dataver
All the dataverse custom functions are also failing to bind on new Fx builds due to #1490.   

Eg, consider a custom function like:

"XSendEmailFromTemplate(GUID(\"88000000-0000-0000-0000-000000000001\"),First(Entity1s),First(Entity1s),[\"valid@contoso.com\"])"

 Before, for arguments that took Dataverse records, they'd specify empty record value.  

: base("XSendEmailFromTemplate", FormulaType.Guid , FormulaType.Guid, RecordType.Empty(), RecordType.Empty(), TableType.Empty())

But with 1490, passing a dataverse record to an empty record is a mismatch (extra fields). 
But we still need some way to allow dataverse records. So allow empty records to accept any type. Need to find a stronger enforcement.

### CODE REVIEW

**Review by LucGenetier — DISMISSED** [review]
:shipit:

File: src/libraries/Microsoft.PowerFx.Interpreter/CustomFunction/CustomTexlFunction.cs [hunk 1]
```diff
@@ -73,11 +73,21 @@ public override bool CheckTypes(CheckTypesContext context, TexlNode[] args, DTyp
...
-                    DType curType = argTypes[i];
+                    DType curType = argTypes[i]; // caller
+                    DType paramType = this.ParamTypes[i]; // callsite 
 
-                    if ((curType.IsRecord || curType.IsTable) && !curType.CheckAggregateNames(this.ParamTypes[i], args[i], errors, SupportCoercionForArg(i), context.Features.PowerFxV1CompatibilityRules))
+                    if (curType.IsRecord || curType.IsTable)
                     {
-                        return false;
+                        // If param type is an empty record, then don't enforce. 
+                        if (paramType.ChildCount == 0)
```
Comment (nguhoa) [inline src/libraries/Microsoft.PowerFx.Interpreter/CustomFunction/CustomTexlFunction.cs:82]: can you do the same fix for private static FormulaValue GetFormulaResult(FormulaValue formulaResult, DType retType) in ReflectionFunction.cs

**Comment by LucGenetier** [comment]
✅ No public API change.

File: src/libraries/Microsoft.PowerFx.Interpreter/CustomFunction/CustomTexlFunction.cs (hunk 1 above)
Comment (nguhoa) [reply src/libraries/Microsoft.PowerFx.Interpreter/CustomFunction/CustomTexlFunction.cs:82]: Fixed.

**Review by nguhoa — APPROVED** [review]
:shipit:

**Comment by LucGenetier** [comment]
✅ No public API change.

**Pull request merged**